code/modules/shuttle/on_move.dm

```diff
@@ -393,13 +393,33 @@ All ShuttleMove procs go here
 /obj/docking_port/stationary/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
 	if(old_dock == src) //Don't move the dock we're leaving
 		return FALSE
+	. = ..()
 
+/obj/docking_port/mobile/fob/beforeShuttleMove(turf/newT, rotation, move_mode)
 	. = ..()
+	previous_dock = get_docked()
+
+/obj/docking_port/mobile/fob/afterShuttleMove(list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir)
+	. = ..()
+	//Keep the planet loaded
+	var/obj/docking_port/stationary/fob/fob_land/destination_dock = get_docked()
+	if(destination_dock.name == "FOB Landing Zone")
+		var/datum/planet/planet = destination_dock.current_planet
+		planet.no_unload_reason = "FOB SHUTTLE"
```

Comment:
you can return after this block since you're obviously not leaving if you just arrived.